lib/Parser/Parse.cpp

```diff
@@ -10473,8 +10473,16 @@ void Parser::FinishDeferredFunction(ParseNodePtr pnodeScopeList)
             pnodeFnc->sxFnc.pnodeVars = nullptr;
             m_ppnodeVar = &pnodeFnc->sxFnc.pnodeVars;
 
+            ParseNodePtr pnodeFncSaveNonLambda = nullptr;
+
+            pnodeFncSaveNonLambda = m_currentNodeNonLambdaFunc;
```

Comment:
On a side note : we don't need to have `GetCurrentNonLamdaFunctionNode `at all. You can always determine that the current function (`funcInfo`) is global or not. We should get rid of this function and underlying variables.